testbed/testbed/mock_backend.go

```diff
@@ -50,7 +50,7 @@ type MockBackend struct {
 
 	// Recording fields.
 	isRecording     bool
-	recordMutex     sync.Mutex
+	RecordMutex     sync.Mutex
```

Comment:
I have a bad feeling it would cause the diff to explode, but instead of exposing the mutex, could we make  the various signal access fields (`ReceivedX`, `DroppedX`, `LogsToRetry`) into functions that use this mutex internally when returning the value?

Reply:
I added a `GetReceivedLogs` function as that is what this PR needs. Other functions could be added as needed. MockBackend hasn't been updated in a very long time and isn't used in any other unit tests in contrib so I don't really wanna spent a lot of time on it.

Reply:
Makes sense to me, looks good.